File: solutions/scala/darts/1/src/main/scala/Darts.scala
object Darts {

  def score(x: Double, y: Double): Int =
    Math.sqrt(Math.pow(x, 2) + Math.pow(y, 2)) match {
      case d if d <= 1  => 10
      case d if d <= 5  => 5
      case d if d <= 10 => 1
      case _            => 0
    }

}

File: solutions/typescript/darts/1/darts.ts
const Circles = {
	inner: { radius: 1, value: 10 },
	middle: { radius: 5, value: 5 },
	outer: { radius: 10, value: 1 },
}

function score(x: number, y: number): number {
	const distanceFromOrigin = Math.sqrt(x ** 2 + y ** 2)
	if (distanceFromOrigin <= Circles.inner.radius) return Circles.inner.value
	if (distanceFromOrigin <= Circles.middle.radius) return Circles.middle.value
	if (distanceFromOrigin <= Circles.outer.radius) return Circles.outer.value
	return 0
}

export { score }
